### DISCUSSION TITLE
Select远程搜索的选择列表和用户预期不同有什么好的解决办法吗？

### DISCUSSION THREAD

**Discussion by xmsz-stu** [DouyinFE/semi-design#2269] [Q&A] [answered]
原先有默认结果
<img width="339" alt="image" src="https://github.com/DouyinFE/semi-design/assets/139225486/84fa00fe-dcf0-492b-aa4e-eac27b35bab1">

我搜索「1」
<img width="325" alt="image" src="https://github.com/DouyinFE/semi-design/assets/139225486/7b3786d3-d7b7-4011-b6f1-9d3c81d66b8b">
然后啥也不选，关闭弹窗，重新打开
<img width="331" alt="image" src="https://github.com/DouyinFE/semi-design/assets/139225486/2a6edb62-0d7c-448c-bd33-3e2a2bfde1bb">
会发现列表还是上次搜索「1」的结果

这时候用户就很懵，因为他并不知道(比较痴呆)是搜索「1」的结果，就会说怎么看不到之前默认结果了...

还有更进一步的情况，比如搜索，然后选择，之后点击清空按钮，然后再打开弹窗，列表还是搜索结果而不是默认列表。但是用户直觉是情况选择+清空搜索。

我看所有组件都没有做处理，但是觉得用户反馈的确实也有道理
特别有个现象，比如我搜索了「1」，然后关闭弹窗，这时候输入框的「1」会被清空，再打开弹窗「1」也没显示，用户根本不知道之前搜索了啥，也不知道之前是不是搜索了。

我的想法

第一个，清空同时清空搜索和选择，对于我们的场景应该是够用的
第二个，打开弹窗的时候，输入框显示之前搜索的结果，这样用户就能知道有没有搜索了
第三个，在列表页显示搜索的关键词+支持清除，好处直观，缺点和输入框的内容重复了

**Comment by pointhalo** [comment, marked as answer]
1. 【清空同时清空搜索和选择】，这个只能由你实现，候选列表由你们控制，我们如果在某些交互下清空列表，就会造成多处input同时影响 optionList，会违背单一控制的原则，这个是不太合理的。你可以直接监听 onClear 然后 更新 optionList = []  即可

2. 
你可以考虑通过  topSlot 自行再强调一下当前搜索关键字
![image](https://github.com/user-attachments/assets/552652b1-fb63-4c4b-9bb4-ec81ffea5745)

或者 searchPosition 设为 dropdown，然后借助它的placeholder在未输入时，显示这个信息也可以
![image](https://github.com/user-attachments/assets/0c98b844-e96c-44da-8527-da219575e1e0)
